awsc/autoscaling/migrate.go

```diff
@@ -264,6 +268,33 @@ func (ms *MigrateService) isECSInstanceDrained(clusterName string, instanceARN s
 	return *instance.Status == ecs.ContainerInstanceStatusDraining && *instance.RunningTasksCount == 0 && *instance.PendingTasksCount == 0, nil
 }
 
+func (ms *MigrateService) isInstanceReady(instance *autoscaling.Instance, clusterName string) (bool, error) {
+	if isHealthyInstance(instance) {
+		if clusterName == "" {
+			return true, nil
+		}
+
+		isRegisteredECSHost, err := ms.isRegisteredECSHost(*instance.InstanceId, clusterName)
```

Comment:
I think you can just return this directly as you've already checked if the instance is healthy
```suggestion
		return ms.isRegisteredECSHost(*instance.InstanceId, clusterName)
```